lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart

```diff
@@ -377,28 +310,26 @@ class HtmlViewEmbedder {
       int viewId = _compositionOrder[i];
 
       if (assertionsEnabled) {
-        if (!_views.containsKey(viewId)) {
+        if (!platformViewManager.knowsViewId(viewId)) {
           debugInvalidViewIds ??= <int>[];
           debugInvalidViewIds.add(viewId);
           continue;
         }
       }
 
       unusedViews.remove(viewId);
-      html.Element platformViewRoot = _rootViews[viewId]!;
+      html.Element platformViewRoot = _viewClipChains[viewId]!.root;
       html.Element overlay = _overlays[viewId]!.htmlElement;
       platformViewRoot.remove();
       skiaSceneHost!.append(platformViewRoot);
       overlay.remove();
       skiaSceneHost!.append(overlay);
       _activeCompositionOrder.add(viewId);
     }
+
     _compositionOrder.clear();
 
-    for (final int unusedViewId in unusedViews) {
-      _releaseOverlay(unusedViewId);
-      _rootViews[unusedViewId]?.remove();
-    }
+    disposeViews(unusedViews);
```

Comment:
I'm not sure this new code will actually dispose of "disposed" views. In the old code we kept a list of views which were disposed, and then later we disposed them. In this code we are only disposing views which are trying to be composited into the scene but were never registered. In the old code we would delete disposed views whether or not they are being actively composited this frame.

Reply:
Ah yes, the thing is that in the old code this was coupled with the creation/destruction of the actual content of the platform view in the DOM.

Now, slots are "lightweight" so they can be destroyed/recreated per frame without impacting performance, and the actual platform view (with the heavy iframe) will still get destroyed when the framework signals a disposal.

Can you think of some example where this code might be leaking unused views?